SharpGen.Runtime/PlatformDetection.cs

```diff
@@ -1,20 +1,12 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Text;
+﻿using System.Runtime.InteropServices;
 
 namespace SharpGen.Runtime
 {
     public static class PlatformDetection
     {
-#if WIN
-        public static bool IsWindows => true;
-        public static bool IsItaniumSystemV => false;
-#elif UNIX
-        public static bool IsWindows => false;
-        public static bool IsItaniumSystemV => true;
-#else
-        public static bool IsWindows => throw null;
-        public static bool IsItaniumSystemV => throw null;
-#endif
+        public static bool IsWindows => RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
```

Comment:
These implementations aren't JIT time constants. Can we calculate these in a static constructor and cache them in a bool field? That will make these more likely to be JIT time constants and get back the perf benefits from the rid-specific implementation.

Reply:
Maybe turn them into `static readonly` fields instead of computed properties with backing fields?

Reply:
Yes lets do that.